### ISSUE TITLE
Better support for wildcard domains

### ISSUE THREAD

**Issue by JonasAlfredsson** [JonasAlfredsson/docker-nginx-certbot#92]
With the latest [pull request](https://github.com/JonasAlfredsson/docker-nginx-certbot/pull/89) we added support for DNS challenges, which means that we could now support wildcard domains. Creating a `server` config which looks like this:

```
server {
    listen              443 ssl;
    server_name         yourdomain.org *.yourdomain.org;
    ssl_certificate_key /etc/letsencrypt/live/test-name/privkey.pem;
    ...
}
```

actually produces a functioning wildcard certificate, but this is not really versatile. A possible solution to this is to make something like this possible:

```
server {
    listen              443 ssl;
    server_name         sub-1.yourdomain.org;
    ssl_certificate_key /etc/letsencrypt/live/yourdomain.org/privkey.pem; #wildcard
    ...
}

server {
    listen              443 ssl;
    server_name         sub-2.yourdomain.org;
    ssl_certificate_key /etc/letsencrypt/live/yourdomain.org/privkey.pem; #wildcard
    ...
}
```

where we pick up the trailing comment and make the certificate request for this certificate only be `-d yourdomain.org -d *.yourdomain.org`. 

We probably need some more logic here to identify and extract when this should happen, and we most likely must demand that the `cert_name` = `yourdomain.org`, so we can skip the tedious thing of trying to guess the "base" domain from the server_name.

**Comment by XaF** [comment]
I'd recommend using the same approach we do for other matches at the moment, with a `.wildcard` or `-wildcard` somewhere in the cert name. As well as a env variable.
The only part where this gets slightly more involved is the "reduction", but that's actually not that complicated to do to handle.

**Comment by XaF** [comment]
Just created #94 that should solve this :)

**Comment by JonasAlfredsson** [comment]
This should now be included in the [v3.0.0](https://github.com/JonasAlfredsson/docker-nginx-certbot/releases/tag/v3.0.0) release.
